File: Plugins/Aspose_Cells_Java_for_PHP/src/aspose/cells/WorkingWithWorksheets/PageSetupFeatures/SettingPageOptions.php
<?php

namespace Aspose\Cells\WorkingWithWorksheets\PageSetupFeatures;

use com\aspose\cells\Workbook as Workbook;
use com\aspose\cells\FileFormatType as FileFormatType;
use com\aspose\cells\PageOrientationType as PageOrientationType;


class SettingPageOptions {

    public static function run($dataDir=null)
    {

        SettingPageOptions::page_orientation($dataDir);

        SettingPageOptions::scaling($dataDir);

    }

    public static function page_orientation($dataDir=null)
    {

        # Instantiating a Workbook object by excel file path
        $workbook = new Workbook();

        # Accessing the first worksheet in the Excel file
        $worksheets = $workbook->getWorksheets();
        $sheet_index = $worksheets->add();
        $sheet = $worksheets->get($sheet_index);

        # Setting the orientation to Portrait
        $page_setup = $sheet->getPageSetup();
        $page_orientation_type = new PageOrientationType();
        $page_setup->setOrientation($page_orientation_type->PORTRAIT);

        # Saving the modified Excel file in default (that is Excel 2003) format
        $workbook->save($dataDir . "Page Orientation.xls");

        print "Set page orientation, please check the output file." . PHP_EOL;
    }

    public static function scaling($dataDir=null)
    {

        # Instantiating a Workbook object by excel file path
        $workbook = new Workbook($dataDir . 'Book1.xls');

        # Accessing the first worksheet in the Excel file
        $worksheets = $workbook->getWorksheets();
        $sheet_index = $worksheets->add();
        $sheet = $worksheets->get($sheet_index);

        # Setting the scaling factor to 100
        $page_setup = $sheet->getPageSetup();
        $page_setup->setZoom(100);

        # Saving the modified Excel file in default (that is Excel 2003) format
        $workbook->save($dataDir . "Scaling.xls");

        print "Set scaling, please check the output file." . PHP_EOL;

    }

}

File: Plugins/Aspose_Cells_Java_for_Jython/asposecells/WorkingWithWorksheets/SettingPageOptions.py
from asposecells import Settings
from com.aspose.cells import Workbook
from com.aspose.cells import PageOrientationType
from java.io import FileInputStream


class SettingPageOptions:

    def __init__(self):
        
        
        self.page_orientation()

        self.scaling()

    def page_orientation(dataDir):
        
        dataDir = Settings.dataDir + 'WorkingWithWorksheets/SettingPageOptions/'

        # Instantiating a Workbook object by excel file path
        workbook = Workbook()

        # Accessing the first worksheet in the Excel file
        worksheets = workbook.getWorksheets()
        sheet_index = worksheets.add()
        sheet = worksheets.get(sheet_index)

        # Setting the orientation to Portrait
        page_setup = sheet.getPageSetup()
        page_orientation_type = PageOrientationType
        page_setup.setOrientation(page_orientation_type.PORTRAIT)

        # Saving the modified Excel file in default (that is Excel 2003) format
        workbook.save(dataDir + "Page Orientation.xls")

        print "Set page orientation, please check the output file."
    
    def scaling(dataDir):
        dataDir = Settings.dataDir + 'WorkingWithWorksheets/SettingPageOptions/'
        
        # Instantiating a Workbook object by excel file path
        workbook = Workbook(dataDir + 'Book1.xls')

        # Accessing the first worksheet in the Excel file
        worksheets = workbook.getWorksheets()
        sheet_index = worksheets.add()
        sheet = worksheets.get(sheet_index)

        # Setting the scaling factor to 100
        page_setup = sheet.getPageSetup()
        page_setup.setZoom(100)

        # Saving the modified Excel file in default (that is Excel 2003) format
        workbook.save(dataDir + "Scaling.xls")

        print "Set scaling, please check the output file."

if __name__ == '__main__':        
    SettingPageOptions()